doc/contribute/documentation/guidelines.rst

```diff
@@ -247,128 +258,183 @@ columns, you can specify ``:widths: 1 2 2``.  If you'd like the browser
+backslash (``\``) before it.
+
+File Names and Commands
+=======================
+
+Sphinx extends reST by supporting additional inline markup elements (called
+"roles") used to tag text with special
+meanings and allow style output formatting. (You can refer to the `Sphinx Inline Markup`_
+documentation for the full list).
+
+While double quotes can be used for rendering text as "code", you are encouraged to use the
+following roles for marking up file names, command names, and other "special" text.
+
+* :rst:role:`file` for file names, e.g., ``:file:`CMakeLists.txt``` will render as
+  :file:`CMakeLists.txt`
+
+  .. note::
+
+     In case you want to indicate a "variable" file path, you may use curly braces to enclose the
+     variable part of the path, e.g., ``:file:`{boardname}_defconfig``` will render as
+     :file:`{boardname}_defconfig`.
+
+* :rst:role:`command` for command names, e.g., ``:command:`make``` will render as :command:`make`
+
+* :rst:role:`envvar` for environment variables, e.g., ``:envvar:`ZEPHYR_BASE``` will render as
+  :envvar:`ZEPHYR_BASE`
+
+For creating references to files that are hosted in the Zephyr organization on GitHub, refer to
+:ref:`linking_to_zephyr_files` section below.
+
+User Interaction
+================
+
+When documenting user interactions, such as key combinations or GUI interactions, use the following
+roles to highlight the commands in a meaningful way:
+
+* :rst:role:`kbd` for keyboard input, e.g., ``:kbd:`Ctrl-C``` will render as :kbd:`Ctrl-C`
+
+* :rst:role:`menuselection` for menu selections, e.g., ``:menuselection:`File --> Open``` will render
+  as :menuselection:`File --> Open`
+
+* :rst:role:`guilabel` for GUI labels, e.g., ``:guilabel:`Cancel``` will render as :guilabel:`Cancel`
+
+Mathematical Formulas
+=====================
+
+You can include mathematical formulas using either the :rst:role:`math` role or :rst:dir:`math`
+directive. The directive provides more flexibility in case you have a more complex formula.
+
+The input language for mathematics is LaTeX markup. Example::
+
+   The answer to life, the universe, and everything is :math:`30 + 2^2 + \sqrt{64} = 42`.
+
+This would render as:
+
+   The answer to life, the universe, and everything is :math:`30 + 2^2 + \sqrt{64} = 42`.
 
 Non-ASCII Characters
-********************
+====================
 
 You can insert non-ASCII characters such as a Trademark symbol (|trade|),
 by using the notation ``|trade|``.
 Available replacement names are defined in an include file used during the Sphinx processing
 of the reST files.  The names of these replacement characters are the same as used in HTML
-entities used to insert characters in HTML, e.g., \&trade; and are defined in the
-file ``sphinx_build/substitutions.txt`` as listed here:
+entities used to insert characters in HTML, e.g., ``\&trade;`` and are defined in the
+file :zephyr_file:`doc/substitutions.txt` as listed below:
```

Comment:
Don't we support UTF-8 natively?

Reply:
I think there is a desire for trying to keep the entire source tree exempt of non-ASCII characters. It might be good to discuss more broadly (ex. Arch WG?) what we think is acceptable, and documentation files could certainly be more "open" than actual source files indeed. But this would still need to be carefully evaluated as this could impact accessibility, would require guidelines w.r.t whether emojis can be used or not, etc.

In any case, I would suggest leaving this section as-is for now (it is carried over from the existing guidelines we already have) and look at addressing UTF-8 guidelines as an enhancement.

Reply:
Just FYI we already got nearly 2000 entries of non-ascii characters in our documentation: 

![image](https://github.com/user-attachments/assets/59922a9d-2652-4aee-a37a-fd278b2ed413)